### PR TITLE
Support for multiple SPI drivers of a given type

### PR DESCRIPTION
**Pull request by bkueng**
Also simplifies the API a bit, since we anyway have to change the drivers to pass additional information (the bus device index) to the lower-level parts. There's a few drivers that are updated to see the differences.
Device ID's are preserved.

fixes https://github.com/PX4/PX4-Autopilot/issues/17247

@dakejahl @dagar 

Any other suggestions while we're at it?

### CODE REVIEW

File: platforms/common/include/px4_platform_common/i2c_spi_buses.h [hunk 1]
```diff
@@ -54,6 +54,39 @@ enum class I2CSPIBusOption : uint8_t {
...
+	spi_mode_e spi_mode;
+	uint32_t spi_devid;
+	int bus_device_index;
+
+	Rotation rotation;
+
+	bool quiet_start;
+	bool keep_running;
+
+	uint8_t orientation;
```
Comment (dagar) [inline platforms/common/include/px4_platform_common/i2c_spi_buses.h:81]: Both rotation and orientation?

File: platforms/common/include/px4_platform_common/i2c_spi_buses.h [hunk 2]
```diff
@@ -102,13 +138,12 @@ class BusCLIArguments
...
 
 	I2CSPIBusOption bus_option{I2CSPIBusOption::All};
-	uint16_t type{0}; ///< device type (driver-specific)
 	int requested_bus{-1};
 	int chipselect_index{1};
 	Rotation rotation{ROTATION_NONE};
 	int bus_frequency{0};
 	spi_mode_e spi_mode{SPIDEV_MODE3};
-	uint8_t i2c_address{0}; ///< optional I2C address: a driver can set this to allow configuring the I2C address
+	uint8_t i2c_address{0}; ///< I2C address (a driver must set the default address)
```
Comment (dagar) [inline platforms/common/include/px4_platform_common/i2c_spi_buses.h:146]: Not necessarily here, but later could you clarify how we should be handling drivers that support a range of I2C addresses?

**Comment by dakejahl** [comment]
Sweet! Thanks for this. I will test this when I get the chance. I don't have the hardware I was using to test at the moment, so it might be a few weeks.

File: platforms/common/include/px4_platform_common/i2c_spi_buses.h (hunk 1 above)
Comment (bkueng) [reply platforms/common/include/px4_platform_common/i2c_spi_buses.h]: Yeah time to remove one

File: platforms/common/include/px4_platform_common/i2c_spi_buses.h (hunk 2 above)
Comment (bkueng) [reply platforms/common/include/px4_platform_common/i2c_spi_buses.h:143]: I see 2 main patterns:
- start multiple independent instances with different addresses
- let the driver handle it internally and cycle through the addresses (so it's not configurable), the mb12xx is an example. If we need more configurability there, we can extend the base, but at this point I don't think it's needed.

**Comment by bkueng** [comment]
Rebased

**Review by dagar — APPROVED** [review]
Looks good, the only difference I came across was the startup output for the Bosch IMU (bmi088_accel & bmi088_gyro instead of bmi088 x 2).

**Pull request merged**